modules/openapi-generator/src/main/resources/powershell/http_signature_auth.mustache

```diff
@@ -1,162 +1,409 @@
-    }
-
-    foreach ($item in $ApiKeysList.GetEnumerator()) {
-        if (![string]::IsNullOrEmpty($item.Name)) {
-            if (Test-Path -Path $item.Value) {
-                $apiKey = $item.Value
-                $apikeyId = $item.Name
-                break;
-            }
-            else {
-                throw "API key file path does not exist."
-            }
-        }
-    }
-
-    if ($ApiKeyPrefixList.ContainsKey($apikeyId)) {
-        $apiPrefix = ApiKeyPrefixList[$apikeyId]
-    }
-
-    if ($apikeyId -and $apiKey -and $apiPrefix) {
-        $result = New-Object -Type PSCustomObject -Property @{ 
-            ApiKeyId       = $apikeyId;
-            ApiKeyFilePath = $apiKey
-            ApiKeyPrefix   = $apiPrefix
-        }
-    }
-    else {
-        return $null
-    }
-    return $result
-}
-
-<#
-.SYNOPSIS
-    Gets the headers for http signed auth.
-
-.DESCRIPTION
-    Gets the headers for the http signed auth. It use (targetpath), date, host and body digest to create authorization header.
+    Gets the headers for the http sigature. 
 .PARAMETER Method
-    Http method
+    HTTP method
 .PARAMETER UriBuilder
     UriBuilder for url and query parameter
 .PARAMETER Body
     Request body
 .OUTPUTS
     Hashtable
 #>
-function Get-{{{apiNamePrefix}}}HttpSignedHeader {
+function Get-{{apiNamePrefix}}HttpSignedHeader {
     param(
         [string]$Method,
         [System.UriBuilder]$UriBuilder,
-        [string]$Body
+        [string]$Body,
+        [hashtable]$RequestHeader
     )
 
+    $HEADER_REQUEST_TARGET = '(request-target)'
+    # The time when the HTTP signature was generated.
+    $HEADER_CREATED = '(created)'
+    # The time when the HTTP signature expires. The API server should reject HTTP requests
+    # that have expired.
+    $HEADER_EXPIRES = '(expires)'
+    # The 'Host' header.
+    $HEADER_HOST = 'Host'
+    # The 'Date' header.
+    $HEADER_DATE = 'Date'
+    # When the 'Digest' header is included in the HTTP signature, the client automatically
+    # computes the digest of the HTTP request body, per RFC 3230.
+    $HEADER_DIGEST = 'Digest'
+    # The 'Authorization' header is automatically generated by the client. It includes
+    # the list of signed headers and a base64-encoded signature.
+    $HEADER_AUTHORIZATION = 'Authorization'
+
     #Hash table to store singed headers
-    $HttpSignedHeader = @{}
+    $HttpSignedRequestHeader = @{ }
+    $HttpSignatureHeader = @{ }
     $TargetHost = $UriBuilder.Host
+    $httpSigningConfiguration = Get-{{apiNamePrefix}}ConfigurationHttpSigning
+    $Digest = $null
+    
+    #get the body digest
+    $bodyHash = Get-{{apiNamePrefix}}StringHash -String $Body -HashName $httpSigningConfiguration.HashAlgorithm
+    if ($httpSigningConfiguration.HashAlgorithm -eq "SHA256") {
+        $Digest = [String]::Format("SHA-256={0}", [Convert]::ToBase64String($bodyHash))
+    }
+    elseif ($httpSigningConfiguration.HashAlgorithm -eq "SHA512") {
```

Comment:
I think elseif should go with the previous line (53 in this case) instead of a new line on its own.